File: code/backend/java/core/src/main/java/games/paths/core/model/story/ClassInfo.java
package games.paths.core.model.story;

import java.util.List;

/**
 * ClassInfo - Domain model for a character class summary.
 * Used within StoryDetail to describe available classes for a story.
 * Immutable record; {@code bonuses} is always a non-null immutable list.
 */
public record ClassInfo(
        Long id,
        String uuid,
        String name,
        String description,
        int weightMax,
        int dexterityBase,
        int intelligenceBase,
        int constitutionBase,
        Integer idCard,
        CardInfo card,
        List<ClassBonusInfo> bonuses) {

    public ClassInfo {
        bonuses = ClassBonusInfoListHelper.immutableCopy(bonuses);
    }
}

File: code/backend/php/src/Core/Domain/Story/ClassInfo.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class ClassInfo
{
    public function __construct(
        public string $uuid,
        public ?string $name = null,
        public ?string $description = null,
        public int $weightMax = 0,
        public int $dexterityBase = 0,
        public int $intelligenceBase = 0,
        public int $constitutionBase = 0
    ) {
    }
}
